### ISSUE TITLE
createHudText doesn't seem to compile

### ISSUE THREAD

**Issue by kayvanbree** [Zezombye/overpy#55]
Hey @Zezombye, trying to paste my compiled code that I just decompiled after you helped me with #54. Overwatch gave me this error:

```
Error: Expected a value after 'heroIconString(' on line 79
```

Line 79:

```
createHudText(eventPlayer, string("{0}: {1}", heroIconString(Reaper), globalVariable(B), null), null, null, Left, 0, White, White, White, VisibleToAndString, Defaultvisibility);
```

I looked into the code that I decompiled and this was in there:

```
Create HUD Text(Event Player, String("{0}: {1}", Hero Icon String(Hero(Reaper)), Global Variable(B), Null), Null, Null, Left, 0,
			White, White, White, Visible To and String, Default Visibility);
```

I now see that heroIconString and globalVariable aren't being compiled either. Is that because the whole line hasn't been compiled?

**Comment by kayvanbree** [comment]
Wait, a lot more seems uncompiled...

Overwatch code:

```
rule("Disable wraith")
{
	event
	{
		Ongoing - Each Player;
		All;
		Reaper;
	}

	actions
	{
		Set Ability 1 Enabled(Event Player, False);
	}
}

rule("Disable shadow step")
{
	event
	{
		Ongoing - Each Player;
		All;
		Reaper;
	}

	actions
	{
		Set Ability 2 Enabled(Event Player, False);
	}
}

rule("Remove life-steal")
{
	event
	{
		Player Dealt Damage;
		All;
		Reaper;
	}

	conditions
	{
		Health(Event Player) != Max Health(Event Player);
		Player Variable(Event Player, I) != Event Player;
	}

	actions
	{
		Damage(Event Player, Event Player, Multiply(Event Damage, 0.300));
		Set Player Variable(Event Player, I, Null);
		Set Player Variable(Event Player, J, Null);
	}
}

rule("Prevent self kill")
{
	event
	{
		Player Took Damage;
		All;
		Reaper;
	}

	conditions
	{
		Attacker != Event Player;
	}

	actions
	{
		Set Player Variable(Event Player, I, Attacker);
		Set Player Variable(Event Player, J, Health(Event Player));
	}
}

rule("Set wraith resource HUD")
{
	event
	{
		Ongoing - Each Player;
		All;
		Reaper;
	}

	actions
	{
		Set Player Variable(Event Player, A, 100);
		Set Player Variable(Event Player, C, 0);
		Set Player Variable(Event Player, K, 0);
		Set Global Variable(A, 50);
		Set Global Variable(B, 40);
		Set Global Variable(C, 30);
		Set Global Variable(D, 3);
		Set Global Variable(E, 150);
		Create HUD Text(Event Player, String("{0}: {1}", String("Form", Null, Null, Null), Player Variable(Event Player, A), Null), Null,
			Null, Left, 0, White, White, White, Visible To and String, Default Visibility);
		Create HUD Text(Event Player, String("{0}: {1}", String("Cooldown", Null, Null, Null), Player Variable(Event Player, C), Null),
			Null, Null, Left, 0, White, White, White, Visible To and String, Default Visibility);
		Create HUD Text(Event Player, String("{0}: {1}", Hero Icon String(Hero(Reaper)), Global Variable(B), Null), Null, Null, Left, 0,
			White, White, White, Visible To and String, Default Visibility);
		Create HUD Text(Event Player, String("{0}: {1}", Hero Icon String(Hero(Reaper)), Global Variable(C), Null), Null, Null, Left, 0,
			White, White, White, Visible To and String, Default Visibility);
		Create HUD Text(Event Player, String("{0}: {1}", Hero Icon String(Hero(Reaper)), Global Variable(D), Null), Null, Null, Left, 0,
			White, White, White, Visible To and String, Default Visibility);
		Create HUD Text(Event Player, String("{0}: {1}", Hero Icon String(Hero(Reaper)), Player Variable(Event Player, B), Null), Null,
			Null, Left, 0, White, White, White, Visible To and String, Default Visibility);
		Create HUD Text(Event Player, String("{0}: {1}", String("Cooldown", Null, Null, Null), Player Variable(Event Player, K), Null),
			Null, Null, Left, 0, White, White, White, Visible To and String, Default Visibility);
	}
}

rule("Wraith form")
{
	event
	{
		Ongoing - Each Player;
		All;
		Reaper;
	}

	conditions
	{
		Player Variable(Event Player, B) == True;
	}

	actions
	{
		Stop Chasing Player Variable(Event Player, C);
		Chase Player Variable At Rate(Event Player, A, 0, Global Variable(B), Destination and Rate);
		Set Move Speed(Event Player, Global Variable(E));
		Set Status(Event Player, Null, Phased Out, 9999);
		Set Primary Fire Enabled(Event Player, False);
		Set Player Variable(Event Player, C, Global Variable(D));
	}
}

rule("Normal form")
{
	event
	{
		Ongoing - Each Player;
		All;
		Reaper;
	}

	conditions
	{
		Player Variable(Event Player, B) == False;
	}

	actions
	{
		Chase Player Variable At Rate(Event Player, A, 100, Global Variable(C), Destination and Rate);
		Set Move Speed(Event Player, 100);
		Set Status(Event Player, Null, Phased Out, 0);
		Set Primary Fire Enabled(Event Player, True);
		Chase Player Variable At Rate(Event Player, C, 0, 1, Destination and Rate);
	}
}

rule("Wraith heal")
{
	event
	{
		Ongoing - Each Player;
		All;
		Reaper;
	}

	conditions
	{
		Player Variable(Event Player, B) == True;
	}

	actions
	{
		Stop Heal Over Time(Player Variable(Event Player, F));
		Start Heal Over Time(Event Player, Event Player, 10000, Global Variable(A));
		Set Player Variable(Event Player, F, Last Heal Over Time ID);
		Disabled Heal(Event Player, Event Player, Player Variable(Event Player, A));
	}
}

rule("Remove wraith heal")
{
	event
	{
		Ongoing - Each Player;
		All;
		Reaper;
	}

	conditions
	{
		Player Variable(Event Player, B) == False;
	}

	actions
	{
		Stop Heal Over Time(Player Variable(Event Player, F));
	}
}

rule("Wraith effect")
{
	event
	{
		Ongoing - Each Player;
		All;
		Reaper;
	}

	conditions
	{
		Player Variable(Event Player, B) == True;
	}

	actions
	{
		Create Effect(All Players(All Teams), Bad Aura, Purple, Add(Eye Position(Event Player), Forward), 2,
			Visible To Position and Radius);
		Set Player Variable(Event Player, D, Last Created Entity);
		Create Effect(All Players(All Teams), Bad Aura Sound, White, Event Player, 300, Visible To Position and Radius);
		Set Player Variable(Event Player, E, Last Created Entity);
	}
}

rule("Stop wraith effect")
{
	event
	{
		Ongoing - Each Player;
		All;
		Reaper;
	}

	conditions
	{
		Player Variable(Event Player, B) == False;
	}

	actions
	{
		Destroy Effect(Player Variable(Event Player, D));
		Destroy Effect(Player Variable(Event Player, E));
	}
}

rule("Start using wraith")
{
	event
	{
		Ongoing - Each Player;
		All;
		Reaper;
	}

	conditions
	{
		Is Button Held(Event Player, Ability 1) == True;
		Is Using Ultimate(Event Player) == False;
		Has Status(Event Player, Hacked) == False;
		Has Status(Event Player, Frozen) == False;
		Has Status(Event Player, Rooted) == False;
		Has Status(Event Player, Knocked Down) == False;
		Has Status(Event Player, Stunned) == False;
		Player Variable(Event Player, C) == 0;
		Player Variable(Event Player, B) == False;
	}

	actions
	{
		Set Player Variable(Event Player, B, True);
	}
}

rule("Stop using wraith")
{
	event
	{
		Ongoing - Each Player;
		All;
		Reaper;
	}

	conditions
	{
		Is Button Held(Event Player, Ability 1) == False;
		Player Variable(Event Player, B) == True;
	}

	actions
	{
		Set Player Variable(Event Player, B, False);
	}
}

rule("Stop using wraith when ulting")
{
	event
	{
		Ongoing - Each Player;
		All;
		Reaper;
	}

	conditions
	{
		Is Using Ultimate(Event Player) == True;
		Player Variable(Event Player, B) == True;
	}

	actions
	{
		Set Player Variable(Event Player, B, False);
	}
}

rule("Stop using wraith when out of juice")
{
	event
	{
		Ongoing - Each Player;
		All;
		Reaper;
	}

	conditions
	{
		Player Variable(Event Player, A) == 0;
		Player Variable(Event Player, B) == True;
	}

	actions
	{
		Set Player Variable(Event Player, B, False);
	}
}

rule("Leap")
{
	event
	{
		Ongoing - Each Player;
		All;
		Reaper;
	}

	conditions
	{
		Is Button Held(Event Player, Ability 2) == True;
		Is Using Ultimate(Event Player) == False;
		Has Status(Event Player, Hacked) == False;
		Has Status(Event Player, Frozen) == False;
		Has Status(Event Player, Rooted) == False;
		Has Status(Event Player, Knocked Down) == False;
		Has Status(Event Player, Stunned) == False;
		Player Variable(Event Player, K) == 0;
		Player Variable(Event Player, B) == False;
	}

	actions
	{
		Set Player Variable(Event Player, K, 3);
		Apply Impulse(Event Player, Facing Direction Of(Event Player), 25, To World, Cancel Contrary Motion);
	}
}

rule("Leap cooldown")
{
	event
	{
		Ongoing - Each Player;
		All;
		Reaper;
	}

	conditions
	{
		Player Variable(Event Player, K) > 0;
	}

	actions
	{
		Chase Player Variable At Rate(Event Player, K, 0, 1, Destination and Rate);
	}
}
```

This is the opy code:

```
@Rule "Disable wraith"
@Event eachPlayer
@Hero reaper
eventPlayer.setAbility1Enabled(false)


@Rule "Disable shadow step"
@Event eachPlayer
@Hero reaper
eventPlayer.setAbility2Enabled(false)


@Rule "Remove life-steal"
@Event playerDealtDamage
@Hero reaper
if eventPlayer.getHealth() != eventPlayer.getMaxHealth() and eventPlayer.I != eventPlayer:
    damage(eventPlayer, eventPlayer, eventDamage * 0.3)
    eventPlayer.I = null
    eventPlayer.J = null


@Rule "Prevent self kill"
@Event playerTookDamage
@Hero reaper
if attacker != eventPlayer:
    eventPlayer.I = attacker
    eventPlayer.J = eventPlayer.getHealth()


@Rule "Set wraith resource HUD"
@Event eachPlayer
@Hero reaper
eventPlayer.A = 100
eventPlayer.C = 0
eventPlayer.K = 0
A = 50
B = 40
C = 30
D = 3
E = 150
hudHeader(eventPlayer, "Form: {}".format(eventPlayer.A), Position.LEFT, 0, Color.WHITE, HudReeval.VISIBILITY_AND_STRING)
hudHeader(eventPlayer, "Cooldown: {}".format(eventPlayer.C), Position.LEFT, 0, Color.WHITE, HudReeval.VISIBILITY_AND_STRING)
hudHeader(eventPlayer, "{}: {}".format(heroIcon(Hero.REAPER), B), Position.LEFT, 0, Color.WHITE, HudReeval.VISIBILITY_AND_STRING)
hudHeader(eventPlayer, "{}: {}".format(heroIcon(Hero.REAPER), C), Position.LEFT, 0, Color.WHITE, HudReeval.VISIBILITY_AND_STRING)
hudHeader(eventPlayer, "{}: {}".format(heroIcon(Hero.REAPER), D), Position.LEFT, 0, Color.WHITE, HudReeval.VISIBILITY_AND_STRING)
hudHeader(eventPlayer, "{}: {}".format(heroIcon(Hero.REAPER), eventPlayer.B), Position.LEFT, 0, Color.WHITE, HudReeval.VISIBILITY_AND_STRING)
hudHeader(eventPlayer, "Cooldown: {}".format(eventPlayer.K), Position.LEFT, 0, Color.WHITE, HudReeval.VISIBILITY_AND_STRING)


@Rule "Wraith form"
@Event eachPlayer
@Hero reaper
if eventPlayer.B:
    stopChasingVariable(eventPlayer.C)
    chase(eventPlayer.A, 0, rate=B, ChaseReeval.DESTINATION_AND_RATE)
    eventPlayer.setMoveSpeed(E)
    eventPlayer.setStatusEffect(null, Status.PHASED_OUT, 9999)
    eventPlayer.setPrimaryFireEnabled(false)
    eventPlayer.C = D


@Rule "Normal form"
@Event eachPlayer
@Hero reaper
if not eventPlayer.B:
    chase(eventPlayer.A, 100, rate=C, ChaseReeval.DESTINATION_AND_RATE)
    eventPlayer.setMoveSpeed(100)
    eventPlayer.setStatusEffect(null, Status.PHASED_OUT, 0)
    eventPlayer.setPrimaryFireEnabled(true)
    chase(eventPlayer.C, 0, rate=1, ChaseReeval.DESTINATION_AND_RATE)


@Rule "Wraith heal"
@Event eachPlayer
@Hero reaper
if eventPlayer.B:
    stopHoT(eventPlayer.F)
    eventPlayer.startHoT(eventPlayer, 10000, A)
    eventPlayer.F = getLastHoT()


@Rule "Remove wraith heal"
@Event eachPlayer
@Hero reaper
if not eventPlayer.B:
    stopHoT(eventPlayer.F)


@Rule "Wraith effect"
@Event eachPlayer
@Hero reaper
if eventPlayer.B:
    createEffect(getAllPlayers(), Effect.BAD_AURA, Color.PURPLE, eventPlayer.getEyePosition() + Vector.FORWARD, 2, EffectReeval.VISIBILITY_POSITION_AND_RADIUS)
    eventPlayer.D = getLastCreatedEntity()
    createEffect(getAllPlayers(), Effect.BAD_AURA_SOUND, Color.WHITE, eventPlayer, 300, EffectReeval.VISIBILITY_POSITION_AND_RADIUS)
    eventPlayer.E = getLastCreatedEntity()


@Rule "Stop wraith effect"
@Event eachPlayer
@Hero reaper
if not eventPlayer.B:
    destroyEffect(eventPlayer.D)
    destroyEffect(eventPlayer.E)


@Rule "Start using wraith"
@Event eachPlayer
@Hero reaper
if eventPlayer.isHoldingButton(Button.ABILITY_1) and not eventPlayer.isUsingUltimate() and not eventPlayer.hasStatusEffect(Status.HACKED) and not eventPlayer.hasStatusEffect(Status.FROZEN) and not eventPlayer.hasStatusEffect(Status.ROOTED) and not eventPlayer.hasStatusEffect(Status.KNOCKED_DOWN) and not eventPlayer.hasStatusEffect(Status.STUNNED) and eventPlayer.C == 0 and not eventPlayer.B:
    eventPlayer.B = true


@Rule "Stop using wraith"
@Event eachPlayer
@Hero reaper
if not eventPlayer.isHoldingButton(Button.ABILITY_1) and eventPlayer.B:
    eventPlayer.B = false


@Rule "Stop using wraith when ulting"
@Event eachPlayer
@Hero reaper
if eventPlayer.isUsingUltimate() and eventPlayer.B:
    eventPlayer.B = false


@Rule "Stop using wraith when out of juice"
@Event eachPlayer
@Hero reaper
if eventPlayer.A == 0 and eventPlayer.B:
    eventPlayer.B = false


@Rule "Leap"
@Event eachPlayer
@Hero reaper
if eventPlayer.isHoldingButton(Button.ABILITY_2) and not eventPlayer.isUsingUltimate() and not eventPlayer.hasStatusEffect(Status.HACKED) and not eventPlayer.hasStatusEffect(Status.FROZEN) and not eventPlayer.hasStatusEffect(Status.ROOTED) and not eventPlayer.hasStatusEffect(Status.KNOCKED_DOWN) and not eventPlayer.hasStatusEffect(Status.STUNNED) and eventPlayer.K == 0 and not eventPlayer.B:
    eventPlayer.K = 3
    eventPlayer.applyImpulse(eventPlayer.getFacingDirection(), 25, Relativity.TO_WORLD, Impulse.CANCEL_CONTRARY_MOTION)


@Rule "Leap cooldown"
@Event eachPlayer
@Hero reaper
if eventPlayer.K > 0:
    chase(eventPlayer.K, 0, rate=1, ChaseReeval.DESTINATION_AND_RATE)
```

The compiled code: 

```
rule ("Disable wraith") {
    event {
        ongoing-eachplayer;
        all;
        Reaper;
    }

    actions {
        setAbility1Enabled(eventPlayer, false);
    }
}

rule ("Disable shadow step") {
    event {
        ongoing-eachplayer;
        all;
        Reaper;
    }

    actions {
        setAbility2Enabled(eventPlayer, false);
    }
}

rule ("Remove life-steal") {
    event {
        playerDealtDamage;
        all;
        Reaper;
    }

    conditions {
        health(eventPlayer) != maxHealth(eventPlayer);
        playerVariable(eventPlayer, I) != eventPlayer;
    }

    actions {
        damage(eventPlayer, eventPlayer, multiply(eventDamage, 0.3));
        setPlayerVariable(eventPlayer, I, null);
        setPlayerVariable(eventPlayer, J, null);
    }
}

rule ("Prevent self kill") {
    event {
        playerTookDamage;
        all;
        Reaper;
    }

    conditions {
        attacker != eventPlayer;
    }

    actions {
        setPlayerVariable(eventPlayer, I, attacker);
        setPlayerVariable(eventPlayer, J, health(eventPlayer));
    }
}

rule ("Set wraith resource HUD") {
    event {
        ongoing-eachplayer;
        all;
        Reaper;
    }

    actions {
        setPlayerVariable(eventPlayer, A, 100);
        setPlayerVariable(eventPlayer, C, 0);
        setPlayerVariable(eventPlayer, K, 0);
        setGlobalVariable(A, 50);
        setGlobalVariable(B, 40);
        setGlobalVariable(C, 30);
        setGlobalVariable(D, 3);
        setGlobalVariable(E, 150);
        createHudText(eventPlayer, string("{0}: {1}", string("form", null, null, null), playerVariable(eventPlayer, A), null), null, null, Left, 0, White, White, White, VisibleToAndString, Defaultvisibility);
        createHudText(eventPlayer, string("{0}: {1}", string("cooldown", null, null, null), playerVariable(eventPlayer, C), null), null, null, Left, 0, White, White, White, VisibleToAndString, Defaultvisibility);
        createHudText(eventPlayer, string("{0}: {1}", heroIconString(Reaper), globalVariable(B), null), null, null, Left, 0, White, White, White, VisibleToAndString, Defaultvisibility);
        createHudText(eventPlayer, string("{0}: {1}", heroIconString(Reaper), globalVariable(C), null), null, null, Left, 0, White, White, White, VisibleToAndString, Defaultvisibility);
        createHudText(eventPlayer, string("{0}: {1}", heroIconString(Reaper), globalVariable(D), null), null, null, Left, 0, White, White, White, VisibleToAndString, Defaultvisibility);
        createHudText(eventPlayer, string("{0}: {1}", heroIconString(Reaper), playerVariable(eventPlayer, B), null), null, null, Left, 0, White, White, White, VisibleToAndString, Defaultvisibility);
        createHudText(eventPlayer, string("{0}: {1}", string("cooldown", null, null, null), playerVariable(eventPlayer, K), null), null, null, Left, 0, White, White, White, VisibleToAndString, Defaultvisibility);
    }
}

rule ("Wraith form") {
    event {
        ongoing-eachplayer;
        all;
        Reaper;
    }

    conditions {
        playerVariable(eventPlayer, B) == true;
    }

    actions {
        stopChasingPlayerVariable(eventPlayer, C);
        chasePlayerVariableAtRate(eventPlayer, A, 0, globalVariable(B), DestinationAndRate);
        setMoveSpeed(eventPlayer, globalVariable(E));
        setStatus(eventPlayer, null, PhasedOut, 9999);
        setPrimaryFireEnabled(eventPlayer, false);
        setPlayerVariable(eventPlayer, C, globalVariable(D));
    }
}

rule ("Normal form") {
    event {
        ongoing-eachplayer;
        all;
        Reaper;
    }

    conditions {
        playerVariable(eventPlayer, B) == false;
    }

    actions {
        chasePlayerVariableAtRate(eventPlayer, A, 100, globalVariable(C), DestinationAndRate);
        setMoveSpeed(eventPlayer, 100);
        setStatus(eventPlayer, null, PhasedOut, 0);
        setPrimaryFireEnabled(eventPlayer, true);
        chasePlayerVariableAtRate(eventPlayer, C, 0, 1, DestinationAndRate);
    }
}

rule ("Wraith heal") {
    event {
        ongoing-eachplayer;
        all;
        Reaper;
    }

    conditions {
        playerVariable(eventPlayer, B) == true;
    }

    actions {
        stopHealOverTime(playerVariable(eventPlayer, F));
        startHealOverTime(eventPlayer, eventPlayer, 10000, globalVariable(A));
        setPlayerVariable(eventPlayer, F, lastHealOverTimeId);
    }
}

rule ("Remove wraith heal") {
    event {
        ongoing-eachplayer;
        all;
        Reaper;
    }

    conditions {
        playerVariable(eventPlayer, B) == false;
    }

    actions {
        stopHealOverTime(playerVariable(eventPlayer, F));
    }
}

rule ("Wraith effect") {
    event {
        ongoing-eachplayer;
        all;
        Reaper;
    }

    conditions {
        playerVariable(eventPlayer, B) == true;
    }

    actions {
        createEffect(allPlayers(AllTeams), BadAura, Purple, add(eyePosition(eventPlayer), forward), 2, VisibleToPositionAndRadius);
        setPlayerVariable(eventPlayer, D, lastCreatedEntity);
        createEffect(allPlayers(AllTeams), BadAuraSound, White, eventPlayer, 300, VisibleToPositionAndRadius);
        setPlayerVariable(eventPlayer, E, lastCreatedEntity);
    }
}

rule ("Stop wraith effect") {
    event {
        ongoing-eachplayer;
        all;
        Reaper;
    }

    conditions {
        playerVariable(eventPlayer, B) == false;
    }

    actions {
        destroyEffect(playerVariable(eventPlayer, D));
        destroyEffect(playerVariable(eventPlayer, E));
    }
}

rule ("Start using wraith") {
    event {
        ongoing-eachplayer;
        all;
        Reaper;
    }

    conditions {
        isButtonHeld(eventPlayer, Ability1) == true;
        isUsingUltimate(eventPlayer) == false;
        hasStatus(eventPlayer, Hacked) == false;
        hasStatus(eventPlayer, Frozen) == false;
        hasStatus(eventPlayer, Rooted) == false;
        hasStatus(eventPlayer, KnockedDown) == false;
        hasStatus(eventPlayer, Stunned) == false;
        playerVariable(eventPlayer, C) == 0;
        playerVariable(eventPlayer, B) == false;
    }

    actions {
        setPlayerVariable(eventPlayer, B, true);
    }
}

rule ("Stop using wraith") {
    event {
        ongoing-eachplayer;
        all;
        Reaper;
    }

    conditions {
        isButtonHeld(eventPlayer, Ability1) == false;
        playerVariable(eventPlayer, B) == true;
    }

    actions {
        setPlayerVariable(eventPlayer, B, false);
    }
}

rule ("Stop using wraith when ulting") {
    event {
        ongoing-eachplayer;
        all;
        Reaper;
    }

    conditions {
        isUsingUltimate(eventPlayer) == true;
        playerVariable(eventPlayer, B) == true;
    }

    actions {
        setPlayerVariable(eventPlayer, B, false);
    }
}

rule ("Stop using wraith when out of juice") {
    event {
        ongoing-eachplayer;
        all;
        Reaper;
    }

    conditions {
        playerVariable(eventPlayer, A) == 0;
        playerVariable(eventPlayer, B) == true;
    }

    actions {
        setPlayerVariable(eventPlayer, B, false);
    }
}

rule ("Leap") {
    event {
        ongoing-eachplayer;
        all;
        Reaper;
    }

    conditions {
        isButtonHeld(eventPlayer, Ability2) == true;
        isUsingUltimate(eventPlayer) == false;
        hasStatus(eventPlayer, Hacked) == false;
        hasStatus(eventPlayer, Frozen) == false;
        hasStatus(eventPlayer, Rooted) == false;
        hasStatus(eventPlayer, KnockedDown) == false;
        hasStatus(eventPlayer, Stunned) == false;
        playerVariable(eventPlayer, K) == 0;
        playerVariable(eventPlayer, B) == false;
    }

    actions {
        setPlayerVariable(eventPlayer, K, 3);
        applyImpulse(eventPlayer, facingDirectionOf(eventPlayer), 25, ToWorld, CancelContraryMotion);
    }
}

rule ("Leap cooldown") {
    event {
        ongoing-eachplayer;
        all;
        Reaper;
    }

    conditions {
        playerVariable(eventPlayer, K) > 0;
    }

    actions {
        chasePlayerVariableAtRate(eventPlayer, K, 0, 1, DestinationAndRate);
    }
}
```

**Comment by Zezombye** [comment]
Hmm, seems the compiler does not add the `Hero()` function, and instead directly adds `Reaper`.

I'll look into it, thanks for the report!

**Comment by Zezombye** [comment]
Fixed in 1.5.

**Comment by kayvanbree** [comment]
Finally got around trying it out. It works!
